### DISCUSSION TITLE
Use with 4.2.2 board

### DISCUSSION THREAD

**Discussion by jeremy-niles** [LeeOtts/Ender3v2-Klipper-Configs#16] [Q&A] [answered]
Hi Lee, 

What would need to be changed to use this package with a 4.2.2 board? Thanks!

**Comment by prodestrian** [comment, marked as answer]
I have an Ender 3 V2 running the stock 4.2.2 board, with a CRTouch and Sprite Extruder.

I was able to install these configuration files just now (with a few features disabled such as KAMP...for now) and can confirm it's looking good so far, I don't actually see any obvious differences between the instructions for 4.2.2 and 4.2.7.

Klipper has successfully connected to my printer and it's showing the current temperatures. Next up I need to run some calibrations, then see if I can get a print going.

Edit: I take that back, as soon as I tried to home the bed nothing happened. Turns out the pins on 4.2.2 are different, as per #8 
I copied over the `step_pin` and `dir_pin` settings from here and that fixed it: https://github.com/Klipper3d/klipper/blob/master/config/printer-creality-ender3-v2-2020.cfg

**Reply by LeeOtts** [reply]
This is the way...  you will just need to change those pins for xyze

**Reply by prodestrian** [reply]
It's been printing great so far! I've performed the bed screw calibration and dialled in my Z-offset, had some slight issues with the `START_PRINT` macro because I forgot to untick the "Emit Temperature commands automatically" setting:
![2024-02-11_08-16](https://github.com/LeeOtts/Ender3v2-Klipper-Configs/assets/34206489/f7084212-310c-4bff-84c6-e63bc0299876)

Also successfully calibrated `pressure_advance` and saw a noticeable improvement in corner quality.

I did make a few changes to the order of operations in your `START_PRINT` macro to match what I had previously but otherwise it's the same. KAMP is working brilliantly too, wouldn't have known about it if I hadn't come across this repo. :tada:

**Reply by LeeOtts** [reply]
Oh, I don't use prusaslicer. I use cura. I will note that in the next update, I am about to push out. 

Yeah I notice significant improvements when I dial in pressure_advance as well too. Have you tried input shaping?

I love KAMP. I always hear great things but thought it would be to difficult to implement when I just started out. I was wrong and it has only been made easier to install.  Kyleisah's hard work and contribution to the Klipper community is truly commendable and makes a valuable impact

**Comment by LeeOtts** [comment]
I added a 4.2.2 board configuration to the repo. Change the file name to printer.cfg 

I don't have a 4.2.2 board, so use it at my own risk. Let me know if it acts funny, and I will make changes, but I just changed pins to match a 4.2.2 config. Everything else is the same, so you must still add the remaining files to work as intended.
